src/commands/deploy.rs

```diff
@@ -505,20 +505,30 @@ impl DeployCommand {
         bindle_version: String,
         app_id: Uuid,
     ) -> Result<Uuid> {
-        let revisions = CloudClient::list_revisions(cloud_client).await?;
-        let revision = revisions
-            .items
-            .iter()
-            .find(|&x| x.revision_number == bindle_version && x.app_id == app_id);
-        Ok(revision
-            .ok_or_else(|| {
-                anyhow!(
-                    "No revision with version {} and app id {}",
-                    bindle_version,
-                    app_id
-                )
-            })?
-            .id)
+        let mut revisions = cloud_client.list_revisions().await?;
+
+        loop {
+            if let Some(revision) = revisions
+                .items
+                .iter()
+                .find(|&x| x.revision_number == bindle_version && x.app_id == app_id)
+            {
+                return Ok(revision.id);
+            }
+
+            if revisions.is_last_page {
+                break;
+            }
+
+            revisions.page_index = revisions.page_index + 1;
+            revisions = cloud_client.list_revisions_next(&revisions).await?;
```

Comment:
should we probably do fn list_revision_page(page_index) instead? changing the previously fetched response does not feel right. But then i am a total noob in Rust, so please ignore me if it does not make sense

Reply:
or may be move the responsibility to increment the page_index within the function

Reply:
The idea of passing the revisions object back was that it could the index and page size and whatever else might needed, the caller didn't need to know.  But if we have to mess with the object then that becomes less attractive.  I think you're right, I called it `previous` so let the callee sort it.

Reply:
Update pushed, could you give it yet another go please?  Thanks!

Reply:
```
Uploading test-app version 0.1.0+rb9f6446c...
Deploying...
total: 107 page_index: 0 page_size: 50 is_last_page: false
total: 107 page_index: 1 page_size: 50 is_last_page: false
total: 107 page_index: 2 page_size: 50 is_last_page: true
```

i think this is working fine now.